From 9c67f377477a07495a712f478eabfa8d33164c45 Mon Sep 17 00:00:00 2001
From: Daniel Gomez <dagmcr@gmail.com>
Date: Sun, 17 Jan 2021 23:32:51 +0100
Subject: [PATCH 1/2] qt: Create subdir in Makefile

Create builddir/qt subdir when --with-qt5 is enabled.

Fixes the following error:

moc -Iqt -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB
-I/var/lib/yocto/gallium/tmp_20210113/work/znver1-poky-linux/zbar/0.23.1+gitAUTOINC+89e7900d85-r0/recipe-sysroot/usr/include/QtWidgets
-I/var/lib/yocto/gallium/tmp_20210113/work/znver1-poky-linux/zbar/0.23.1+gitAUTOINC+89e7900d85-r0/recipe-sysroot/usr/include/QtGui
-I/var/lib/yocto/gallium/tmp_20210113/work/znver1-poky-linux/zbar/0.23.1+gitAUTOINC+89e7900d85-r0/recipe-sysroot/usr/include/QtCore
-I../git/include ../git/include/zbar/QZBar.h -o qt/moc_QZBar.cpp
moc: Cannot create qt/moc_QZBar.cpp
make: *** [Makefile:2459: qt/moc_QZBar.cpp] Error 1

Signed-off-by: Daniel Gomez <dagmcr@gmail.com>
---
 qt/Makefile.am.inc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/qt/Makefile.am.inc b/qt/Makefile.am.inc
index b099c2b..c98fbeb 100644
--- a/qt/Makefile.am.inc
+++ b/qt/Makefile.am.inc
@@ -15,4 +15,5 @@ qt/moc_%.cpp: qt/%.h
 	$(MOC) $(qt_libzbarqt_la_CPPFLAGS) $< -o $@
 
 qt/moc_%.cpp: include/zbar/%.h
+	@$(mkdir_p) qt 2>/dev/null
 	$(MOC) $(qt_libzbarqt_la_CPPFLAGS) $< -o $@
-- 
2.29.2

